grr/server/grr_response_server/flows/general/osquery_test.py

```diff
@@ -155,17 +160,18 @@ def setUp(self):
     super(FakeOsqueryFlowTest, self).setUp()
     self.client_id = self.SetupClient(0)
 
-  def _InitializeFlow(self, query):
+  def _InitializeFlow(self, query, file_collect_columns=[]):
```

Comment:
Mutable instances should not be used when specifying default parameters as they are global to the entire program. The correct way is something like this:

```python
def _InitializeFlow(self, query: str, file_collect_columns: Optional[List[str]] = None):
    if file_collect_columns is None:
        file_collect_columns = []
    (...)
```

Reply:
Good catch, didn't know that!

Reply:
I decided to remove the default value altogether and pass an empty list explicitly in the other uses. I found this clearer, and adding `[]` to a few places doesn't clutter the code too much, but it will help in debugging in case there are errors. Please comment if you disagree.